usaspending_api/search/v2/views/spending_by_subaward_grouped.py

```diff
@@ -0,0 +1,150 @@
+                "key": "sort",
+                "type": "enum",
+                "enum_values": ["award_id", "subaward_count", "award_generated_internal_id", "subaward_obligation"],
+                "text_type": "search",
+                "default": "award_id",
+            },
+        ]
+
+        # Accepts the same filters as spending_by_award
+        self.models.extend(copy.deepcopy(AWARD_FILTER_NO_RECIPIENT_ID))
+        self.models.extend(copy.deepcopy([model for model in PAGINATION if model["name"] != "sort"]))
+
+    @cache_response()
+    def post(self, request: Request) -> Response:
+        """Return all subawards matching given awards"""
+        self.original_filters = request.data.get("filters")
+        json_request = self.validate_request_data(request.data)
+        filters = json_request.get("filters", {})
+        self.pagination = {
+            "limit": json_request["limit"],
+            "lower_bound": (json_request["page"] - 1) * json_request["limit"],
+            "page": json_request["page"],
+            "sort_key": json_request.get("sort"),
+            "sort_order": json_request["order"],
+            "upper_bound": json_request["page"] * json_request["limit"] + 1,
+        }
+
+        time_period_obj = SubawardSearchTimePeriod(
+            default_end_date=settings.API_MAX_DATE, default_start_date=settings.API_SEARCH_MIN_DATE
+        )
+
+        query_with_filters = QueryWithFilters(QueryType.SUBAWARDS)
+        filter_query = query_with_filters.generate_elasticsearch_query(filters=filters, options=time_period_obj)
+        results = self.build_elasticsearch_search_with_aggregation(filter_query)
+
+        return Response(self.construct_es_response(results))
+
+    def validate_request_data(self, request_data: Request) -> dict[str, Any]:
+
+        tiny_shield = TinyShield(self.models)
+        return tiny_shield.block(request_data)
+
+    def construct_es_response(self, results: list[SubawardGroupedModel]) -> dict[str, Any]:
+        return {
+            "limit": self.pagination["limit"],
+            "results": results,
+            "page_metadata": {
+                "page": self.pagination["page"],
+                "hasNext": True if len(results) > self.pagination["limit"] else False,
+            },
+            "messages": get_generic_filters_message(
+                self.original_filters.keys(), [elem["name"] for elem in self.models]
+            ),
+        }
+
+    def build_elasticsearch_search_with_aggregation(self, filter_query: ES_Q) -> dict[str, Any]:
+        # Aggregate the ES query to group the subaward values by their prime award
+        terms_aggregation = A("terms", field="award_piid_fain")
+
+        # Get the unique_award_key for each prime award
+        terms_aggregation.metric("unique_award_key", "terms", field="unique_award_key")
+
+        # Sum the subaward amount within each prime award
+        terms_aggregation.metric("subaward_obligation", "sum", field="subaward_amount")
```

Comment:
We have to do some special handling of float sums with Elasticsearch otherwise you get value such as below. You can find examples in the different spending_by_ endpoints and spending_over_time.
```
{
            "award_id": "NNM17AA12C",
            "subaward_count": 228,
            "award_generated_internal_id": "CONT_AWD_NNM17AA12C_8000_-NONE-_-NONE-",
            "subaward_obligation": 28947569.990000002
        },
```